### PR TITLE
Move build_connect_params to a class method

### PR DESCRIPTION
**Pull request by agrare**
The UI calls raw_connect and needs to build the connect_params.  Instead
of duplicating code in the UI we can move this to a class method similar
to auth_url which can be called from the UI.

### CODE REVIEW

**Comment by bronaghs** [comment]
@agrare - Is there a related UI PR?

**Comment by agrare** [comment]
@bronaghs yes https://github.com/ManageIQ/manageiq-ui-classic/pull/2575

File: app/models/manageiq/providers/microsoft/infra_manager.rb [hunk 1]
```diff
@@ -37,12 +37,37 @@ def self.auth_url(hostname, port = nil)
...
+  end
+
   def connect(options = {})
     raise "no credentials defined" if self.missing_credentials?(options[:auth_type])
 
     hostname           = options[:hostname] || self.hostname
     options[:endpoint] = self.class.auth_url(hostname, port)
-    connect_params     = build_connect_params(options)
+    options[:user]   ||= authentication_userid(options[:auth_type])
+    options[:password] = MiqPassword.try_decrypt(options[:password]) || authentication_password(options[:auth_type])
```
Comment (bronaghs) [inline app/models/manageiq/providers/microsoft/infra_manager.rb]: @agrare - you are calling `MiqPassword.try_decrypt...` twice.

Comment (agrare) [reply app/models/manageiq/providers/microsoft/infra_manager.rb]: Good catch

Comment (agrare) [reply app/models/manageiq/providers/microsoft/infra_manager.rb]: Three times if you include raw_connect (it was already doing it twice).
I'll change it to just decrypt in raw_connect.

Comment (agrare) [reply app/models/manageiq/providers/microsoft/infra_manager.rb]: Fixed, tested credential validation and refresh and both work

**Comment by jntullo** [comment]
LGTM, thanks for fixing this @agrare  🎉

**Comment by djberg96** [comment]
LGTM, though I'm confused by the bug tag.

**Comment by agrare** [comment]
@djberg96 currently you can't validate credentials therefore you can't add an SCVMM provider using the UI

**Comment by djberg96** [comment]
@agrare Huh? That must be recent. Ok. Anyway, approved. :)

**Comment by agrare** [comment]
@bronaghs @djberg96 travis is green

**Comment by agrare** [comment]
> That must be recent.

Yeah within the past week

**Comment by simaishi** [comment]
Gaprindashvili backport details:

```
$ git log -1
commit f13f6e59c2e17ada0495b8a0f286f3c615d23c24
Author: Daniel Berger <djberg96@gmail.com>
Date:   Tue Oct 31 11:59:54 2017 -0600

    Merge pull request #34 from agrare/move_build_connect_params_to_class_method
    
    Move build_connect_params to a class method
    (cherry picked from commit 72c7a5a07d1f99f0b541605ce7f431fc08a0878e)
    
    https://bugzilla.redhat.com/show_bug.cgi?id=1510071
```

**Pull request merged**